clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/MetricsDuringTopicCreationDeletionTest.java

```diff
@@ -0,0 +1,194 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients;
+
+import org.apache.kafka.common.test.ClusterInstance;
+import org.apache.kafka.common.test.api.ClusterConfigProperty;
+import org.apache.kafka.common.test.api.ClusterTest;
+import org.apache.kafka.common.test.api.Type;
+import org.apache.kafka.server.config.ReplicationConfigs;
+import org.apache.kafka.server.config.ServerConfigs;
+import org.apache.kafka.server.config.ServerLogConfigs;
+import org.apache.kafka.server.metrics.KafkaYammerMetrics;
+
+import com.yammer.metrics.core.Gauge;
+
+import org.junit.jupiter.api.BeforeEach;
+
+import java.io.Closeable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MetricsDuringTopicCreationDeletionTest {
+
+    private static final String TOPIC_NAME_PREFIX = "topic";
+    private static final int TOPIC_NUM = 2;
+    private static final int CREATE_DELETE_ITERATIONS = 3;
+    private static final int REPLICATION_FACTOR = 1;
+    private static final int PARTITION_NUM = 3;
+
+    private final ClusterInstance clusterInstance;
+    private final List<String> topics;
+    private volatile boolean running = true;
+
+    private int initialOfflinePartitionsCount = 0;
+    private int initialPreferredReplicaImbalanceCount = 0;
+    private int initialUnderReplicatedPartitionCount = 0;
+
+    public MetricsDuringTopicCreationDeletionTest(ClusterInstance clusterInstance) {
+        this.clusterInstance = clusterInstance;
+        this.topics = new ArrayList<>();
+        for (int n = 0; n < TOPIC_NUM; n++) {
+            topics.add(TOPIC_NAME_PREFIX + n);
+        }
+    }
+
+    /*
+     * Captures initial values of key controller metrics.
+     * These will be compared with final values to detect any changes.
+     */
+    @BeforeEach
+    public void setUp() {
+        try {
+            initialOfflinePartitionsCount = getGauge("OfflinePartitionsCount").value();
+            initialPreferredReplicaImbalanceCount = getGauge("PreferredReplicaImbalanceCount").value();
+            initialUnderReplicatedPartitionCount = getGauge("UnderReplicatedPartitions").value();
```

Comment:
Will these initial values get other value after `setUp` function? If not, we can remove this block and set them as final value.

```java
    private final int initialOfflinePartitionsCount = 0;
    private final int initialPreferredReplicaImbalanceCount = 0;
    private final int initialUnderReplicatedPartitionCount = 0;
```